### PR TITLE
Performance Calculator GUI

### PR DESCRIPTION
**Pull request by stanriders**
image
image

### CODE REVIEW

**Comment by peppy** [comment]
I've done a basic clean-up pass. Lots more I'd like to do, but going to leave that for the future.

I think this is a pretty cool direction – can see pieces of this being integrated into the game to give players insight on their plays / beatmaps.

**Review by peppy — APPROVED** [review]

File: PerformanceCalculatorGUI/Screens/ProfileScreen.cs [hunk 1]
```diff
@@ -250,12 +250,15 @@ private void calculateProfile(string username)
                     }
                 });
 
-                int index = 0;
-                decimal totalLocalPP = (decimal)localOrdered.Select(x=> x.PP).Sum(play => Math.Pow(0.95, index++) * play);
+                decimal totalLocalPP = 0;
+                for (var i = 0; i < localOrdered.Count; i++)
+                    totalLocalPP += (decimal)(Math.Pow(0.95, i++) * (localOrdered[i].PP ?? 0));
```
Comment (smoogipoo) [inline PerformanceCalculatorGUI/Screens/ProfileScreen.cs:255]: I... What? :psyduck:

Comment (peppy) [reply PerformanceCalculatorGUI/Screens/ProfileScreen.cs:255]: Would you please elaborate? I am not seeing what you are.

Comment (smoogipoo) [reply PerformanceCalculatorGUI/Screens/ProfileScreen.cs:255]: The double increment. It's fine I'll fix.

Comment (peppy) [reply PerformanceCalculatorGUI/Screens/ProfileScreen.cs:255]: aha, nice catch

**Pull request merged**